### PR TITLE
feat: new error types for destinations (tiktok_ads, statsig, splitio, snowflake) 

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by koladilip — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

File: src/v0/destinations/splitio/transform.js [hunk 1]
```diff
@@ -82,10 +86,14 @@ function prepareResponse(message, destination, category) {
...
       default:
-        throw new CustomError("Message type not supported", 400);
+        throw new InstrumentationError(
+          `Event type ${message.type} is not supported`
+        );
     }
   } else {
-    throw new CustomError("eventTypeId does not match with ideal format", 400);
+    throw new InstrumentationError(
+      "eventTypeId does not match with ideal format"
```
Comment (utsabc) [inline src/v0/destinations/splitio/transform.js:95]: Lets paste the regex in the error message itself

File: src/v0/destinations/statsig/transform.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 } = require("../../util");
 const { ENDPOINT } = require("./config");
+const { InstrumentationError } = require("../../util/errorTypes");
 
 function process(event) {
   const { message, destination } = event;
 
   if (!message.type) {
-    throw new CustomError("invalid message type for statsig", 400);
+    throw new InstrumentationError("invalid message type for statsig");
```
Comment (utsabc) [inline src/v0/destinations/statsig/transform.js:13]:
```suggestion
    throw new InstrumentationError("Event type not present");
```

File: src/v0/destinations/tiktok_ads/transform.js [hunk 3]
```diff
@@ -85,9 +88,8 @@ const getTrackResponse = (message, Config, event) => {
           phone_number.trim()
         ).toString();
       } else {
-        throw new CustomError(
-          "Invalid phone number. Include proper country code except +86 and the phone number length must be no longer than 15 digit. Aborting",
-          400
+        throw new InstrumentationError(
+          "Invalid phone number. Include proper country code except +86 and the phone number length must be no longer than 15 digit. Aborting"
```
Comment (utsabc) [inline src/v0/destinations/tiktok_ads/transform.js:92]: Maybe share regex for reference or the link to tiktok docs ?

File: src/v0/destinations/tiktok_ads/transform.js [hunk 4]
```diff
@@ -113,13 +115,13 @@ const trackResponseBuilder = async (message, { Config }) => {
...
   if (!event) {
-    throw new CustomError("Event name is required", 400);
+    throw new InstrumentationError("Event name is required");
   }
 
   const standardEventsMap = getHashFromArrayWithDuplicate(eventsToStandard);
 
   if (eventNameMapping[event] === undefined && !standardEventsMap[event]) {
-    throw new CustomError(`Event name (${event}) is not valid`, 400);
+    throw new InstrumentationError(`Event name (${event}) is not valid`);
```
Comment (utsabc) [inline src/v0/destinations/tiktok_ads/transform.js:124]:
```suggestion
    throw new InstrumentationError(`Event name (${event}) is not valid, must be mapped to one of standard events`);
```

File: src/v0/destinations/tiktok_ads/transform.js [hunk 5]
```diff
@@ -143,18 +145,15 @@ const process = async event => {
   const { message, destination } = event;
 
   if (!destination.Config.accessToken) {
-    throw new CustomError("Access Token not found. Aborting ", 400);
+    throw new ConfigurationError("Access Token not found. Aborting ", 400);
```
Comment (utsabc) [inline src/v0/destinations/tiktok_ads/transform.js:148]:
```suggestion
    throw new ConfigurationError("Access Token not found. Aborting");
```

**Pull request merged**